modules/rdp_proxy/rdp_proxy_linux_amd64.go

```diff
@@ -240,53 +314,70 @@ func (mod *RdpProxy) Configure() (err error) {
 func (mod *RdpProxy) handleRdpConnection(payload *nfqueue.Payload) int {
     // 1. Determine source and target addresses.
     p := gopacket.NewPacket(payload.Data, layers.LayerTypeIPv4, gopacket.Default)
-    src, sport := p.NetworkLayer().NetworkFlow().Src(), p.TransportLayer().TransportFlow().Src()
-    dst, dport := p.NetworkLayer().NetworkFlow().Dst(), p.TransportLayer().TransportFlow().Dst()
+    src, sport := p.NetworkLayer().NetworkFlow().Src().String(), fmt.Sprintf("%s", p.TransportLayer().TransportFlow().Src())
+    dst, dport := p.NetworkLayer().NetworkFlow().Dst().String(), fmt.Sprintf("%s", p.TransportLayer().TransportFlow().Dst())
 
-    ips := fmt.Sprintf("[%v:%v -> %v:%v]", src, sport, dst, dport)
+    // TODO : Log everything inside the events stream
+    ips := fmt.Sprintf("[%s:%s -> %s:%s]", src, sport, dst, dport)
 
-    if mod.isTarget(dst.String()) {
-        target := fmt.Sprintf("%v:%v", dst, dport)
+    if mod.isTarget(dst) {
+        target := fmt.Sprintf("%s:%s", dst, dport)
 
         // 2. Check if the destination IP already has a PyRDP session active, if so, do nothing.
         if _, ok :=  mod.active[target]; !ok {
-            // 3.1. Otherwise, create a proxy agent and firewall rules.
-            args := []string{
-                "-l", fmt.Sprintf("%d", mod.startPort),
-                // "-o", mod.outpath,
-                // "-i", "-d"
-                target,
-            }
-
-            //   3.2. Spawn PyRDP proxy instance
-            cmd := exec.Command(mod.cmd, args...)
-            stderrPipe, _ := cmd.StderrPipe()
-
-            if err := cmd.Start(); err != nil {
-                // Wont't handle things like "port already in use" since it happens at runtime
-                mod.Error("PyRDP Start error : %v", err.Error())
-                mod.Info("Failed to start PyRDP, won't intercept target %s", ips)
-
-                // Add an exception in the firewall to avoid intercepting packets to this destination and port
-                mod.doReturn(dst.String(), dport, true)
-                payload.SetVerdict(nfqueue.NF_DROP)
-
-                return 0
+            targetNLA, _ := mod.isNLAEnforced(target)
+
+            // Only if seccheck is set
+            if targetNLA {
+                switch mod.nlaMode {
+                case "RELAY":
+                    mod.Info("%s Target has NLA enabled and mode RELAY, unimplemented", ips)
```

Comment:
Maybe make this check at plugin startup (when parsing options) before letting the user run in relay mode